Adding NFC API

pmorau opened the pull request:
This adds the Webinos NFC API.
Included is an Android implementation and a simulated
API implementation for testing on other platforms.

pmorau committed:
Adding NFC API

This adds the Webinos NFC API.
Included is an Android implementation and a simulated
API implementation for testing on other platforms.

paddybyers commented:
Per, this hasn't been landed yet because some other instability emerged towards the end of last week. When we think it's stable again we should rebase this.

pmorau committed:
Updating listener model in Android NFC

With this change Android modules will set themselves as single
listener for tag events and the dispatching of events to actual
listeners will happen entirely on js side.

pmorau committed:
Adding read/write and Ndef push

The read/write from/to tag and Ndef push via
Android beam has been added.

pmorau committed:
Remove short types from NdefRecord

Modified the NdefRecord to contain no short types since bridge
cannot handle it. This so that NdefRecord arrays can be
transferred during write and share tag operations.
